app/javascript/components/patient/laboratory/LaboratoryTable.js

```diff
@@ -0,0 +1,438 @@
+            Lab Results
+            <InfoTooltip tooltipTextKey="labResults" location="right" className="pl-1" />
+          </Card.Header>
+          <Card.Body className="my-1">
+            <Row className="mb-4">
+              <Col>
+                <Button onClick={this.toggleAddModal}>
+                  <i className="fas fa-plus fa-fw"></i>
+                  <span className="ml-2">Add New Lab Result</span>
+                </Button>
+              </Col>
+              <Col xl={6} lg={10} md={12}>
+                <InputGroup size="md" className="mt-3 mt-md-0">
+                  <InputGroup.Prepend>
+                    <OverlayTrigger overlay={<Tooltip>Search by ID, Lab Type, or Result.</Tooltip>}>
+                      <InputGroup.Text className="rounded-0">
+                        <i className="fas fa-search"></i>
+                        <label htmlFor="laboratories-search-input" className="ml-1 mb-0" aria-label="Search Laboratories Table by ID, Lab Type, or Result.">
+                          Search
+                        </label>
+                      </InputGroup.Text>
+                    </OverlayTrigger>
+                  </InputGroup.Prepend>
+                  <Form.Control
+                    id="laboratories-search-input"
+                    autoComplete="off"
+                    size="md"
+                    name="search"
+                    onChange={this.handleSearchChange}
+                    aria-label="Search"
+                  />
+                </InputGroup>
+              </Col>
+            </Row>
+            <CustomTable
+              dataType="laboratories"
+              columnData={this.state.table.colData}
+              rowData={this.state.table.rowData}
+              totalRows={this.state.table.totalRows}
+              handleTableUpdate={query => this.updateTable({ ...this.state.query, order: query.orderBy, page: query.page, direction: query.sortDirection })}
+              handleEntriesChange={this.handleEntriesChange}
+              loading={this.state.loading}
+              page={this.state.query.page}
+              handlePageUpdate={this.handlePageUpdate}
+              entryOptions={this.state.entryOptions}
+              entries={this.state.query.entries}
+              orderBy={this.state.query.order}
+              sortDirection={this.state.query.direction}
+              tableCustomClass="table-has-dropdown"
+            />
+          </Card.Body>
+        </Card>
+        {(this.state.showAddModal || this.state.showEditModal) && (
+          <LaboratoryModal
+            currentLabData={this.state.showAddModal ? {} : this.getCurrLab()}
+            onClose={this.state.showAddModal ? this.toggleAddModal : this.toggleEditModal}
+            onSave={this.state.showAddModal ? this.handleAddSubmit : this.handleEditSubmit}
+            editMode={this.state.showEditModal}
+            loading={this.state.loading}
+            only_positive_lab={this.isOnlyPosLab()}
```

Comment:
Nit: only_positive_lab is snake case instead of camel case here.

Reply:
this prop originates from the show erb file where we use snake case and we keep them the same even when they trickle to the react components (ex: `edit_mode`, `authenticity_token`, `jurisdiction_paths`, `assigned_users`, etc)